public/app/features/dashboard/components/ShareModal/SharePublicDashboardUtils.ts

```diff
@@ -1,21 +1,47 @@
 import { getBackendSrv } from '@grafana/runtime';
+import { notifyApp } from 'app/core/actions';
+import { createErrorNotification, createSuccessNotification } from 'app/core/copy/appNotification';
 import { DashboardModel } from 'app/features/dashboard/state';
+import { dispatch } from 'app/store/store';
+import { DashboardDataDTO, DashboardMeta } from 'app/types/dashboard';
 
 export interface PublicDashboardConfig {
   isPublic: boolean;
+  publicDashboard: {
+    uid: string;
+    dashboardUid: string;
+    orgId: number;
+    timeSettings?: object;
+  };
+}
+export interface DashboardResponse {
+  dashboard: DashboardDataDTO;
+  meta: DashboardMeta;
 }
 
 export const dashboardCanBePublic = (dashboard: DashboardModel): boolean => {
   return dashboard?.templating?.list.length === 0;
 };
 
-export const getPublicDashboardConfig = async (dashboardUid: string) => {
+export const getDashboard = async (dashboardUid: string, setDashboard: Function) => {
```

Comment:
we usually type function callbacks, but think we can refactor this to use useReducer and thinks later, will simplify a lot

Reply:
I typed the function for now. I'll make a note of possible refactor.